oidc-service-impl/src/main/java/io/mosip/esignet/services/OAuthServiceImpl.java

```diff
@@ -115,8 +122,13 @@ public TokenResponse getTokens(TokenRequestV2 tokenRequest, String dpopHeader, b
 
         ClientDetail clientDetailDto = clientManagementService.getClientDetails(transaction.getClientId());
 
-        tokenService.verifyClientAssertionToken(clientDetailDto.getId(), clientDetailDto.getPublicKey(), tokenRequest.getClient_assertion(),
-                isV2? (String) oauthServerDiscoveryMap.get("token_endpoint") : discoveryIssuerId+"/oauth/token");
+        List<String> validAudience;
+        if (isV2) {
+            validAudience = List.of((String) oauthServerDiscoveryMap.get(PAR_ENDPOINT), (String) oauthServerDiscoveryMap.get(TOKEN_ENDPOINT), (String) discoveryMap.get(ISSUER));
```

Comment:
Please remove PAR from this list, it should only have token and issuer-identifier